src/Workspaces/Core/Portable/Workspace/Host/TemporaryStorage/TrivialTemporaryStorageService.TrivialStorageStreamHandle.cs

```diff
@@ -12,11 +12,15 @@ internal sealed partial class TrivialTemporaryStorageService
 {
     private sealed class TrivialStorageStreamHandle(
         TemporaryStorageIdentifier storageIdentifier,
-        StreamStorage streamStorage) : ITemporaryStorageStreamHandle
+        MemoryStream streamCopy) : ITemporaryStorageStreamHandle
     {
         public TemporaryStorageIdentifier Identifier => storageIdentifier;
 
         public Stream ReadFromTemporaryStorage(CancellationToken cancellationToken)
-            => streamStorage.ReadStream();
+        {
+            // Return a read-only view of the underlying buffer to prevent users from overwriting or directly
+            // disposing the backing storage.
+            return new MemoryStream(streamCopy.GetBuffer(), 0, (int)streamCopy.Length, writable: false);
```

Comment:
>new MemoryStream [](http://example.com/codeflow?start=19&length=16)

Does this method not get called on many of the TrivialStorageStreamHandle instance created? If it's called on most/all of them, it seems like this could just be stored in the field.

If the earlier comment about combining the interfaces into ITemporaryStorageHandle were used, and if the caller would just pass in an appopriate read only stream, then the stream/text versions of these classes could be combined too.

Reply:
I'm preserving the behavior from before.  I'm genuinely not sure if we have a single scenario exercising this code in the real world, so I'd prefer to leave the semantics as is for safety.